include/wil/registry.h

```diff
@@ -17,16 +17,2704 @@
+         * \exception std::exception (including wil::ResultException) will be thrown on all failures, including value not found
+         */
+        inline ::std::wstring get_value_string(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name)
+        {
+            return ::wil::reg::get_value<::std::wstring>(key, subkey, value_name);
+        }
+        template <>
+        inline ::std::wstring get_value_string<::std::wstring>(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name)
+        {
+            return ::wil::reg::get_value<::std::wstring>(key, subkey, value_name);
+        }
+
+        /**
+         * \brief Reads a REG_SZ value, returning a std::wstring
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param value_name A string specifying the name of the registry value to read from
+         *        can be nullptr to read from the unnamed default registry value
+         * \return A std::wstring created from the string value read from the registry
+         *         note, the returned string will have already passed through ExpandEnvironmentStringsW
+         * \exception std::exception (including wil::ResultException) will be thrown on all failures, including value not found
+         */
+        inline ::std::wstring get_value_string(HKEY key, _In_opt_ PCWSTR value_name)
+        {
+            return ::wil::reg::get_value<::std::wstring>(key, nullptr, value_name);
+        }
+        template <>
+        inline ::std::wstring get_value_string<::std::wstring>(HKEY key, _In_opt_ PCWSTR value_name)
+        {
+            return ::wil::reg::get_value<::std::wstring>(key, nullptr, value_name);
+        }
+
+        /**
+         * \brief Reads a REG_EXPAND_SZ value, returning a std::wstring
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param subkey A string to append to the HKEY to attempt to read from
+         *        can be nullptr if not needed
+         * \param value_name A string specifying the name of the registry value to read from
+         *        can be nullptr to read from the unnamed default registry value
+         * \return A std::wstring created from the string value read from the registry
+         *         note, the returned string will have already passed through ExpandEnvironmentStringsW
+         * \exception std::exception (including wil::ResultException) will be thrown on all failures, including value not found
+         */
+        inline ::std::wstring get_value_expanded_string(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name)
+        {
+            ::std::wstring value;
+            const reg_view_details::reg_view regview{ key };
+            regview.get_value(subkey, value_name, value, REG_EXPAND_SZ);
+            return value;
+        }
+        template <>

```

Comment:
nit, add blank line here between functions

Reply:
yeah. fixed that + put a doxygen comment on that function